openid-4-verifiable-presentations-1_0.md

```diff
@@ -320,6 +320,24 @@ The following is a non-normative example of a transaction data content, after ba
 }
 ```
 
+`attachments`:
+: OPTIONAL. Array of objects, where each object includes an extra resource that needs to be passed to the Wallet. These resources are intended to provide additional context to the Wallet or the End-User. The Wallet MUST ignore any unrecognized attachment formats. Each object contains the following parameters:
+
+    * `format`: REQUIRED: String that identifies the format of the attachment. The specific values are out of scope of this specification. It is RECOMMENDED to use collision-resistant names for type values.
+    * `data`: REQUIRED: The structure and content of data are format-dependent and not constrained by this specification.
+    * `credential_ids`: OPTIONAL. Array of strings each referencing a Credential requested by the Verifier for which the attachment is relevant. Each string matches the `id` field in a DCQL Credential Query. If omitted the attachment is relevant to all requested credentials.
```

Comment:
Isn't credential_ids already very specific to a particular attachment format? Is there a specific reason to define this as part of the base structure instead of leaving it to the format specific data field?

Reply:
In case of an authorization credential, you need to tell the Wallet to which credential it is mapped, because there is no direct link from the authorization credential to the requested credential.
It seems to be a better approach than looping over all credentials to check if there is at least one that could match